app/pages/6_Costs.py

```diff
@@ -0,0 +1,188 @@
+    res = st.selectbox(
+        "Resolution",
+        choices,
+        format_func=lambda x: choices[x], 
+        key="gen_res",
+        help="You can choose a resolution for time aggregation applied for a plot"
+    )
+    st.markdown(fix_cursor_css, unsafe_allow_html=True)
+
+buses_country_data = gen_buses_dict_list.get(selected_network)
+gen_buses_df = buses_country_data["p"].drop("Load", axis=1, errors="ignore")
+
+_, date_range_param, _ = st.columns([1, 50, 1])
+with date_range_param:
+    min_index = gen_buses_df.index[0]
+    max_index = gen_buses_df.index[-1]
+    min_value = datetime.datetime(min_index.year, min_index.month, min_index.day)
+    max_value = datetime.datetime(max_index.year, max_index.month, max_index.day)
+    values = st.slider(
+            "Select a range of values",
+            min_value, max_value, (min_value, max_value),
+            # step=datetime.timedelta(hours=int(res[:-1])),
+            format="D MMM, HH:mm",
+            label_visibility="hidden",
+            key="gen_date"
+        )
+
+with carrier_col:
+    carrier = st.selectbox(
+        "Costs for...",
+        ["electricity", "gas"],
+        help="You can choose the costs of a distinct carrier."
+    )
+
+countries_codes = pd.unique(
+    [(lambda x: re.sub("\d.*", '', x))(x) for x in gen_buses_df.columns]
+)
+# to avoid mis-interpretaiton of regex outputs
+country_codes_clean = [x for x in countries_codes if x in helper.config["countries_names"].keys()]
+country_codes_clean.insert(0, "all")
+with country_col:
+    ctr = st.selectbox(
+        "Country",
+        country_codes_clean,
+        format_func=country_formatter, 
+        key="country",
+        help="You can choose a country to examine operation of the energy system"
+    )
+    st.markdown(fix_cursor_css, unsafe_allow_html=True)  
+
+# ###################### electricity costs #####################
+
+fig = sp.make_subplots(
+        rows=2, cols=1,
+    )
+
+costs_dict_list = helper.get_marginal_costs_dict(ctr)
+costs_weighted_dict_list = helper.get_weighted_costs_dict()
+
+for selected_network in selected_networks:
+
+    res_h = str(res) + "H"
+
+    costs_unagg = costs_dict_list.get(selected_network)
+    costs = costs_unagg[carrier].loc[values[0]:values[1]].squeeze().resample(res_h).mean()
+
+    fig.add_trace(
+        go.Scatter(x=costs.index, y=costs.values,
+        mode='lines', name=f"{scenario_formatter(selected_network)}"), row=1, col=1
+    )
+
+    costs_weighted = costs_weighted_dict_list.get(selected_network)
+    costs = costs_weighted[carrier].sort_values()
+
+    new_trace = go.Bar(
+        x=costs.index, y=costs.values,
+        textposition='auto', name=f"{scenario_formatter(selected_network)}"
+    )
+    fig.add_trace(new_trace, row=2, col=1)
+
+    fig.update_yaxes(title_text='Costs [EUR/MWh]', row=1, col=1)
+    fig.update_yaxes(title_text='Average costs [EUR/MWh]', row=2, col=1)
+
+    fig.update_layout(width=1000, height=600)
+
+st.plotly_chart(fig)
```

Comment:
The plot format currently differs from others. I think, it could make sense to keep everything consistent. Application of the plot parameters like here https://github.com/ekatef/heat-peaks-lit/blob/8231892645066dd84a6a2bcfe49d439ec2bfcae1/app/pages/3_Country_operation.py#L52-L68 should do the trick. Probably, we may also move these definitions to the config.

Reply:
With this I'm not sure what to do, I like the style of the plot actually